src/libmeasurement_kit/ooni/facebook_messenger.cpp

```diff
@@ -0,0 +1,221 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "private/common/fcompose.hpp"
+#include "private/common/utils.hpp"
+#include "private/ooni/utils.hpp"
+#include "private/ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+static void dns_many(Error error,
+                     Var<Entry> entry,
+                     std::map<std::string, std::string> fb_service_hostnames,
+                     Settings options,
+                     Var<Reactor> reactor,
+                     Var<Logger> logger,
+                     Callback<Error,
+                              Var<Entry>,
+                              std::map<std::string, std::vector<std::string>>,
+                              Settings,
+                              Var<Reactor>,
+                              Var<Logger>> cb
+                    ) {
+    Var<std::map<std::string, std::vector<std::string>>>
+        fb_service_ips(new std::map<std::string, std::vector<std::string>>(
+            {{"stun", {}},
+             {"b_api", {}},
+             {"b_graph", {}},
+             {"edge", {}},
+             {"external_cdn", {}},
+             {"scontent_cdn", {}},
+             {"star", {}}}
+        ));
+
+    if (error) {
+        cb(error, entry, *fb_service_ips, options, reactor, logger);
+        return;
+    }
+
+    // if we find any inconsistent DNS later, switch this to true
+    (*entry)["facebook_dns_blocking"] = false;
+
+    int names_count = fb_service_hostnames.size();
+    Var<int> names_tested(new int(0));
+
+    if (names_count == *names_tested) {
```

Comment:
@joelanders I still don't get the logic of this `if`: you've just created `names_tested` to be zero. Is the objective to check whether `names_count` is zero?

Reply:
ah whoops, i changed one location and not the other